# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [64]:
import numpy as np

In [66]:
f = lambda x: x**3

def calcular_area(a: int, b: int, func: callable, n_ret: int) -> float:
    delta = (b - a) / n_ret  
    x_cols = np.linspace(a, b - delta, n_ret) + delta / 2 
    y_cols = func(x_cols) 

    print(f'Número de retângulos = {n_ret}')
    print(f'Delta = {delta}')

    return sum(delta * y_cols) 

In [68]:
calcular_area(a=1, b=2, func=f, n_ret=4)

Número de retângulos = 4
Delta = 0.25


3.7265625

In [70]:
calcular_area(a=1, b=2, func=f, n_ret=100)

Número de retângulos = 100
Delta = 0.01


3.7499624999999996

# 2. Calculando de forma analítica

Resolvendo primeiramente sem os limites de integração:

$$
F(y) = \int f(x^3) \, dx
$$

Integrando \( f(x^3) \):

$$
F(y) = \int x^3 \, dx = \frac{1}{3+1} \cdot x^{3+1} + C
$$

Simplificando:

$$
F(y) = \frac{1}{4} \cdot x^{4} + C
$$

Aplicando os limites de integração de 1 a 2:

$$
F(y) = \int_1^2 x^3 \, dx = F(2) - F(1)
$$

Substituindo os valores:

$$
F(2) = \frac{1}{4} \cdot 2^{4} = \frac{16}{4} = 4
$$

$$
F(1) = \frac{1}{4} \cdot 1^{4} = \frac{1}{4}
$$

Calculando a diferença:

$$
F(2) - F(1) = 4 - \frac{1}{4} = \frac{16}{4} - \frac{1}{4} = \frac{15}{4}
$$

O valor da área é:

**$$ 3.75 $$**